application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateService.java

```diff
@@ -0,0 +1,79 @@
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.service;
+
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.APPROVED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.REJECTED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.COMPLETED;
+
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.revalidation.dto.RecommendationStatusCheckDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.entity.Recommendation;
+import uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+@Slf4j
+@Service
+public class RecommendationTisStatusUpdateService {
+
+  @Autowired
+  private RecommendationRepository recommendationRepository;
+  @Autowired
+  private DoctorsForDBRepository doctorsForDBRepository;
+
+  public void updateRecommendationAndTisStatus(
+      final RecommendationStatusCheckDto recommendationStatusCheckDto) {
+
+    String gmcReferenceNumber = recommendationStatusCheckDto.getGmcReferenceNumber();
+    Optional<DoctorsForDB> optionalDoctorsForDB = doctorsForDBRepository
+        .findById(gmcReferenceNumber);
+    Optional<Recommendation> optionalRecommendation = recommendationRepository
+        .findById(recommendationStatusCheckDto.getRecommendationId());
+
+    RecommendationGmcOutcome recommendationGmcOutcome = recommendationStatusCheckDto.getOutcome();
+
+    if (optionalRecommendation.isPresent() && optionalRecommendation.get().getGmcNumber()
+        .equals(gmcReferenceNumber)) {
+      final var recommendation = optionalRecommendation.get();
+
+      //if gmc outcome is approved/rejected, update the relevant recommendation to approved/rejected, tis status to complete
+      if (recommendationGmcOutcome.name().equals(APPROVED.name()) || recommendationGmcOutcome.name()
+          .equals(REJECTED.name())) {
+        recommendation.setOutcome(recommendationGmcOutcome);
+        recommendation.setRecommendationStatus(COMPLETED);
+      }
+
+      //also update the tis status in doctorsfordb to complete
+      if (optionalDoctorsForDB.isPresent()) {
+        final var doctorsForDB = optionalDoctorsForDB.get();
+        doctorsForDB.setDoctorStatus(COMPLETED);
+        doctorsForDBRepository.save(doctorsForDB);
+      }
+      recommendationRepository.save(recommendation);
```

Comment:
Would this need to fall into the if cause in Line 64 `if (recommendationGmcOutcome.name().equals(APPROVED.name()) || recommendationGmcOutcome.name()` and if the original outcome is not same as the new outcome?